service/service_setup.go

```diff
@@ -33,22 +33,23 @@ import (
 	"github.com/pkg/errors"
 )
 
-func (svc *serviceCore) setupLogging(cfg config.Provider) {
+func (svc *serviceCore) setupLogging() {
 	if svc.log == nil {
-		logBuilder := ulog.Builder()
-		// load and configure logging
+		logBuilder := ulog.Builder().WithScope(svc.metrics)
+
 		err := svc.configProvider.Get("logging").PopulateStruct(&svc.logConfig)
 		if err != nil {
 			ulog.Logger().Info("Logging configuration is not provided, setting to default logger", "error", err)
 		}
+
 		svc.log = logBuilder.WithConfiguration(svc.logConfig).Build()
 	} else {
 		svc.log.Debug("Using custom log provider due to service.WithLogger option")
 	}
 }
 
-func (svc *serviceCore) setupStandardConfig(cfg config.Provider) error {
-	if err := cfg.Get(config.Root).PopulateStruct(&svc.standardConfig); err != nil {
+func (svc *serviceCore) setupStandardConfig() error {
```

Comment:
Mostly simplifying for three reasons:

1. the parameters being passed in are totally redundant on a method call.
2. most of these functions were not even using the parameters and used the object directly
3. needed access to the metrics scope in the logging function and rather than adding an additional parameter, simplified the whole approach.

I would be ok to restoring parameter passing, but in that case I think these methods should be changed to function and take service as a parameter. Kind of like the options pattern